Menu.cpp

```diff
@@ -7,7 +7,24 @@ const char SALIR = '3';
 
 using namespace std;
 
+Menu::Menu(const string& rutaArchAeropuertos, const string& rutaArchVuelos) {
+    Cargador cargador;
+    Diccionario<string, Aeropuerto*> *pDiccionario;
+    Diccionario<string, Aeropuerto*> DiccionarioOut;
+    pDiccionario = &DiccionarioOut;
+    cargador.cargarDatos(pDiccionario, rutaArchAeropuertos);
+
+    this->menuAeropuerto.asignarArbolAeropuertos(pDiccionario);
+
+    mostrarOpciones();
+
+
+    //hacer esto mismo pero con el grafo cuando este lista la clase.
+}
```

Comment:
Crearía el diccionario en el heap, y al método `mostrarOpciones()` lo llamaría desde el `main()`
```suggestion
Menu::Menu(const string& rutaArchAeropuertos, const string& rutaArchVuelos) {
    Cargador cargador;
    Diccionario<string, Aeropuerto*> *pDiccionario = new Diccionario<string, Aeropuerto*>;
    cargador.cargarDatos(pDiccionario, rutaArchAeropuertos);
    this->menuAeropuerto.asignarArbolAeropuertos(pDiccionario);
}
```
Ojo que si hacemos esto vamos a tener que liberar esa memoria, para eso podemos:
- Agregar un método en `MenuAeropuertos` y `MenuVuelos` que se llame `liberarRecursos()` y se encarguen de hacer el `delete aeropuertos;` y `delete vuelos;` y crear un destructor en `Menu` que llame a esos métodos
- Agregar un destructor en `MenuAeropuertos` y `MenuVuelos` que se encarguen de hacer el `delete aeropuertos;` y `delete vuelos;`

Reply:
Para mi no es necesario el diccionario en el heap , porque solamente tiene un puntero a la raíz del árbol.

Reply:
Alex me cometo que si el diccionario esta en el stack, y llama al metodo mostrarMenu() desde fuera del constructor, el diccionario estaba lleno de basura. No entendi bien por qué pasa eso ni tampoco lo probe a ver que onda, pero una solución seria guardarlo en memoria dinamica.

Reply:
Ah okey , entonces si pasa eso probemos de la otra manera

Reply:
Ahh, ahí me acorde. Lo que pasa es que al crearlo en el stack solo existe dentro del constructor. Entonces cuando sale del constructor y vuelve al main, en donde se llamaria al método mostrarOpciones(), se perdió el diccionario

Reply:
Ah listo , entonces no Alex , el `mostrarOpciones` va afuera y hagamos lo que dijo Valen

Reply:
Que genialidad es los debates por acá :)

Reply:
Cual fue la resolucion de esto ? Creeria que si pasa lo que dice Valen, la solucion tambien es la que plantea ella.